config/routes.rb

```diff
@@ -20,6 +20,9 @@
     end
 
     resource :sessions, only: [:new, :destroy] do
+      get :authn_urls, on: :collection
+      get :multifactor, on: :member
+      get :identity_proof, on: :member
```

Comment:
For my own knowledge, can you describe the Vets.gov use-cases for these urls?  I kind of thought ID.me handled all of this stuff on their own.

Reply:
+1 I _think_ I understand it all but it would be good to document all the variations that are available now in terms of how users in different scenarios map to a sequence of API calls.

Reply:
Obviously not as a gate to merging, but in the near future.

Reply:
I will add comments in sessions_controller not here in routes.rb, but yes, let me elaborate.

One of the main things to understand is this new concept around "connect." ID.me uses this to delineate which originating policy needs to do the multifactor or identity proofing at any given time. It essentially triggers a flow on their end, for potentially shared authn_contexts. Let me elaborate further...

Imagine `/sessions/new` as completely deprecated. The new front-end will NEVER call it. Instead it will call `auth_urls`, which returns 3 urls fo 1) mhv sign-in, 2) dslogon sign-in, and 3) idme LOA 1 as you currently know it (or the equivalent to `/sessions/new?level=1`).

If you need to identity proof a LOA1 account that you know has highest attained LOA3, or if you need to level up an LOA1 account to LOA3 that has never been leveled up before (NOTE: this could happen for mhv, dslogon, or idme), you have to already have a vets.gov session. Hence why identity_proof endpoint is now a separate endpoint and requires authentication (doesn't skip it). You will get a 401 not authorized if you try to fetch that url without providing a valid session token.

Finally, multifactor. This is for users who have previously opted out of MFA when signing up for the first time. This could be dslogon, mhv, or even possibly idme (though in the case of idme i think it is maybe still required for LOA3). In any case, if user has previously opted out, they can add it to their account at a later time. When they visit their /users/account page they will see a yellow banner that tells them to consider adding it for additional security. The link for triggering adding it is generated by this `/sessions/multifactor` path. Since the `authn_context` is 'multifactor' we're required to pass a `connect=<originating_authn_context>` to set up the `multifactor` for the right flow.